docs/migration-guides/v4.md

```diff
@@ -0,0 +1,396 @@
+
+If you're defining a "placeholder hook" so that you have a custom Tool Kit command you can run, **this is no longer required**; commands can be defined in your repo's `.toolkitrc.yml`. A placeholder hook will look something like this:
+
+```js
+const { Hook } = require('@dotcom-tool-kit/types')
+
+class Placeholder extends Hook {
+  check() {
+    return true
+  }
+  install() {}
+}
+
+exports.hooks = {
+  'custom:command': Placeholder
+}
+```
+
+With Tool Kit v4, you no longer need to define this command as a hook to run it; [specifying commands that run tasks](#11-rename-hooks-to-commands) in your `.toolkitrc.yml` is all that's needed. Your placeholder hook can be deleted.
+
+### 2.4. Plugins with hooks extending `PackageJsonHook` or `CircleCIConfigHook`
+
+Instead of needing to define new `package.json` scripts or CircleCI jobs in plugins, you can now do this via hook configuration options in your repo's main `.toolkitrc.yml`.
+
+A custom `package.json` hook might look like this:
+
+```js
+const { PackageJsonHook } = require('@dotcom-tool-kit/package-json-hook')
+
+class DemoPublishHook extends PackageJsonHook {
+    constructor() {
+        super(...arguments)
+        this.key = 'demo-publish'
+        this.hook = 'demo:publish'
+    }
+}
+
+exports.hooks = { 'demo:publish': DemoPublishHook }
+```
+
+This manages `package.json` to add a `scripts["demo-publish"]` that calls `dotcom-tool-kit demo:publish`. To replicate this, add this configuration in your repo's `.toolkitrc.yml`:
+
+```yml
+options:
+  hooks:
+    - PackageJson:
+        scripts:
+          demo-publish: 'demo:publish'
+```
+
+A custom `.circleci/config.yml` hook might look like this:
+
+```js
+const CircleCiConfigHook = require('@dotcom-tool-kit/circleci/lib/circleci-config');
+const { TestCI } = require('@dotcom-tool-kit/circleci/lib/index');
+
+class DeployProduction extends CircleCiConfigHook.default {
+	constructor () {
+		super(...arguments);
+		this.job = 'tool-kit/deploy-production';
+		this.jobOptions = {
+			requires: [new TestCI(this.logger).job],
+			filters: { branches: { only: 'main' } }
+		};
+	}
+}
+
+exports.hooks = {
+	'deploy:production': DeployProduction
+}
+```
+
+This adds a `tool-kit/deploy-production` job that depends on the `tool-kit/test-ci` job. To replicate this, add this configuration in your repo's `.toolkitrc.yml`:
+
+```yml
+options:
+  hooks:
+    - CircleCi:
+        jobs:
+          - name: deploy-production
+            command: 'deploy:production'
+        workflows:
+          - name: 'tool-kit'
+            jobs:
+              - name: 'deploy-production'
+                requires:
+                  - 'test'
+                custom:
+                  filters:
+                    branches:
+                      only: main
+```
+
+### 2.5. Plugins with hooks doing other things
+
+You'll need to [talk to the Platforms team](https://financialtimes.enterprise.slack.com/archives/C02TRE2V2Q1) to get support with your migration.
```

Comment:
again: nobody actually has any hooks doing other things. mostly i'm including this because looking like we're covering all the bases will give people more confidence